### Часть 1: Загрузка и первичный осмотр данных (обязательно)

#### 1.1. Загрузите данные

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('data_cts_prisons_and_prisoners.csv', sep=';')

#### 1.2. Общая информация о данных

In [3]:
# Размерность данных
print(f"Количество строк: {df.shape[0]}")
print(f"Количество столбцов: {df.shape[1]}")

# Первые строки
print(df.head())

# Последние строки
print(df.tail())

# Информация о типах данных и пропусках
print(df.info())

# Названия всех столбцов
print(df.columns.tolist())

Количество строк: 84025
Количество столбцов: 13
  Iso3_code                            Country    Region  \
0       BRB                           Barbados  Americas   
1       CRI                         Costa Rica  Americas   
2       DMA                           Dominica  Americas   
3    GBR_NI  United Kingdom (Northern Ireland)    Europe   
4       ITA                              Italy    Europe   

                         Subregion                Indicator  \
0  Latin America and the Caribbean  Persons entering prison   
1  Latin America and the Caribbean  Persons entering prison   
2  Latin America and the Caribbean  Persons entering prison   
3                  Northern Europe  Persons entering prison   
4                  Southern Europe  Persons entering prison   

           Dimension              Category    Sex    Age  Year  \
0  by selected crime  Intentional Homicide  Total  Total  2016   
1  by selected crime  Intentional Homicide  Total  Total  2016   
2  by selected

In [4]:
df.duplicated().sum()

np.int64(0)


| Показатель | Значение |
|------------|----------|
| Количество наблюдений (строк) |84025|
| Количество переменных (столбцов) |13|
| Размер данных в памяти (MB) |8.3+|
| Количество дубликатов строк |0|

#### 1.3. Типы данных

In [7]:
# Типы данных
dtypes_df = pd.DataFrame({
    'Название переменной': df.columns,
    'Тип в pandas': df.dtypes.values,
    'Количество уникальных значений': [df[col].nunique() for col in df.columns]
})
print(dtypes_df)

    Название переменной Тип в pandas  Количество уникальных значений
0             Iso3_code     category                             227
1               Country     category                             227
2                Region     category                               5
3             Subregion       object                              17
4             Indicator     category                               6
5             Dimension     category                               7
6              Category     category                              35
7                   Sex     category                               3
8                   Age     category                               3
9                  Year        int64                              27
10  Unit of measurement     category                               3
11                VALUE      float64                           45002
12               Source       object                             113


|Название переменной|Тип в pandas|Количество уникальных значений|
|---|---|---|
|Iso3_code|object|227|
|Country|object|227|
|Region|object|5|
|Subregion|object|17|
|Indicator|object|6|
|Dimension|object|7|
|Category|object|35|
|Sex|object|3|
|Age|object|3|
|Year|int64|27|
|Unit of measurement|object|3|
|VALUE|float64|45002|
|Source|object|113|

In [6]:
df['Iso3_code'] = df['Iso3_code'].astype('category')
df['Country'] = df['Country'].astype('category')
df['Region'] = df['Region'].astype('category')
df['Subregion'] = df['Unit of measurement'].astype('category')
df['Indicator'] = df['Indicator'].astype('category')
df['Dimension'] = df['Dimension'].astype('category')
df['Category'] = df['Category'].astype('category')
df['Sex'] = df['Sex'].astype('category')
df['Age'] = df['Age'].astype('category')
df['Unit of measurement'] = df['Unit of measurement'].astype('category')
df['Unit of measurement'] = df['Unit of measurement'].astype('category')
